src/xor.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2022 Jan Holthuis
+//
+// This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy
+// of the MPL was not distributed with this file, You can obtain one at
+// http://mozilla.org/MPL/2.0/.
+//
+// SPDX-License-Identifier: MPL-2.0
+
+//! Wrapper types for dealing with XOR encryption.
+
+use binrw::io::{Read, Result, Seek, SeekFrom, Write};
+use std::iter::Cycle;
+
+/// Stream cipher wrapper around another [`Read`]- or [`Write`]-able stream that XOR's all bytes
+/// after reading/before writing.
+#[derive(Debug)]
+pub struct XorStream<T> {
+    stream: T,
+    key: Cycle<std::vec::IntoIter<u8>>,
+    key_size: u64,
```

Comment:
since there is no way to create a key that is larger than `usize`, I don't understand why the key_size is `u64`

Reply:
Because we only need the `key_size` for the `Seek` trait, and [`seek()` uses `u64`](https://doc.rust-lang.org/std/io/trait.Seek.html#tymethod.seek). We could store a `usize`, but then we'd have to convert each time seek is called.